tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/encoder/RegularDataEncoder.java

```diff
@@ -0,0 +1,438 @@
+ * each data point in this block in order to checkout whether there are missing points exist in the data.
+ * If there is, create a bitmap for this block to denote the position of missing points. Next, store
+ * the data info (the data size, the minimum delta value and the first data point of this block) and the
+ * bitmap with its info into the result byte array output stream.</p>
+ *
+ * @author tsunghantsai
+ */
+public abstract class RegularDataEncoder extends Encoder {
+
+  protected static final int BLOCK_DEFAULT_SIZE = 128;
+  private static final Logger LOGGER = LoggerFactory.getLogger(RegularDataEncoder.class);
+  protected ByteArrayOutputStream out;
+  protected int blockSize;
+
+  protected int writeIndex = -1;
+
+  /**
+   * constructor of RegularDataEncoder.
+   *
+   * @param size - the number how many numbers to be packed into a block.
+   */
+  public RegularDataEncoder(int size) {
+    super(TSEncoding.REGULAR);
+    blockSize = size;
+  }
+
+  protected abstract void writeHeader() throws IOException;
+
+  protected abstract void reset();
+
+  protected abstract void flushBlockBuffer(ByteArrayOutputStream out) throws IOException;
+
+  protected void writeHeaderToBytes() throws IOException {
+    out.write(BytesUtils.intToBytes(writeIndex));
+    writeHeader();
+  }
+
+  /**
+   * calling this method to flush all values which haven't encoded to result byte array.
+   */
+  @Override
+  public void flush(ByteArrayOutputStream out) {
+    try {
+      flushBlockBuffer(out);
+    } catch (IOException e) {
+      LOGGER.error("flush data to stream failed!", e);
+    }
+  }
+
+  public static class IntRegularEncoder extends RegularDataEncoder {
+
+    private int[] data;
+    private int[] missingPointData;
+    private int[] regularData;
+    private int firstValue;
+    private int previousValue;
+    private int minDeltaBase;
+    private boolean isMissingPoint;
+    private boolean isLastPack;
+    private BitSet bitmap;
+
+    public IntRegularEncoder() {
+      this(BLOCK_DEFAULT_SIZE);
+    }
+
+    /**
+     * constructor of RegularDataEncoder.
+     *
+     * @param size - the number how many numbers to be packed into a block.
+     */
+    public IntRegularEncoder(int size) {
+      super(size);
+      reset();
+    }
+
+    @Override
+    protected void flushBlockBuffer(ByteArrayOutputStream out) throws IOException {
+      if (writeIndex == -1) {
+        return;
+      }
+
+      this.out = out;
+      // write last pack
+      if (writeIndex < blockSize) {
+        isLastPack = true;
+        checkMissingPoint(out);
+      }
+      // write identifier
+      out.write(BytesUtils.boolToBytes(isMissingPoint));
+      // write bitmap if missing points exist
+      if (isMissingPoint) {
+        writeBitmap(out);
+      }
+      // write header
+      writeHeaderToBytes();
+
+      reset();
+      writeIndex = -1;
+    }
+
+    @Override
+    protected void reset() {
+      blockSize = BLOCK_DEFAULT_SIZE;
```

Comment:
Whatever value `blockSize` is assigned in the constructor function, it will be reset to `BLOCK_DEFAULT_SIZE`, which makes this parameter **meaningless**.

Reply:
Done. Thx.